### PR TITLE
PHOENIX-7029: Add support for multiple query services in PhoenixTestDriver

### PR DESCRIPTION
**Pull request by palashc**

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/jdbc/PhoenixTestDriverIT.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package org.apache.phoenix.jdbc;
```
Comment (shahrs87) [inline phoenix-core/src/it/java/org/apache/phoenix/jdbc/PhoenixTestDriverIT.java:1]: Apache license header missing.

File: phoenix-core/src/it/java/org/apache/phoenix/jdbc/PhoenixTestDriverIT.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+@Category(NeedsOwnMiniClusterTest.class)
+public class PhoenixTestDriverIT extends BaseTest {
+
+    @BeforeClass
+    public static synchronized void doSetup() throws Exception {
+        Map<String, String> props = Maps.newHashMapWithExpectedSize(1);
+        setUpTestDriver(new ReadOnlyProps(props.entrySet().iterator()));
+    }
+
```
Comment (shahrs87) [inline phoenix-core/src/it/java/org/apache/phoenix/jdbc/PhoenixTestDriverIT.java:33]: Can you add one more test where you create 2 urls with the same URL, create 2 different connections but test that it returns the same CQSI? 

Can you add one more test where you create 2 urls with the different URL (with different principals), create 2 different connections but test that it returns different CQSI?

File: phoenix-core/src/test/java/org/apache/phoenix/jdbc/PhoenixTestDriver.java
Comment (palashc) [inline phoenix-core/src/test/java/org/apache/phoenix/jdbc/PhoenixTestDriver.java:1]: @jpisaac FYI

**Comment by palashc** [comment]
Test Report - https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1664/4/testReport/

File: phoenix-core/src/it/java/org/apache/phoenix/jdbc/PhoenixTestDriverIT.java [hunk 3]
```diff
@@ -31,13 +49,48 @@ public static synchronized void doSetup() throws Exception {
...
+    public void testSameCQSI() throws SQLException {
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+        String url = QueryUtil.getConnectionUrl(props, config, "client1");
+        try (Connection conn1 = DriverManager.getConnection(url);
+             Connection conn2 = DriverManager.getConnection(url)) {
+            ConnectionQueryServices cqs1 = conn1.unwrap(PhoenixConnection.class).getQueryServices();
+            ConnectionQueryServices cqs2 = conn2.unwrap(PhoenixConnection.class).getQueryServices();
+            Assert.assertNotNull(cqs1);
+            Assert.assertNotNull(cqs2);
+            Assert.assertTrue("Connections using the same URL should have the same CQSI object.", cqs1.equals(cqs2));
```
Comment (shahrs87) [inline phoenix-core/src/it/java/org/apache/phoenix/jdbc/PhoenixTestDriverIT.java:65]: You could have used assertEquals also.

**Review by shahrs87 — APPROVED** [review]

File: phoenix-core/src/it/java/org/apache/phoenix/jdbc/PhoenixTestDriverIT.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+        String url1 = QueryUtil.getConnectionUrl(props, config, "client1");
+        String url2 = QueryUtil.getConnectionUrl(props, config, "client2");
+        try (Connection conn1 = DriverManager.getConnection(url1);
+             Connection conn2 = DriverManager.getConnection(url2)) {
+            ConnectionQueryServices cqs1 = conn1.unwrap(PhoenixConnection.class).getQueryServices();
+            ConnectionQueryServices cqs2 = conn2.unwrap(PhoenixConnection.class).getQueryServices();
+            Assert.assertNotNull(cqs1);
+            Assert.assertNotNull(cqs2);
+            Assert.assertTrue("Connections using different URL should have different CQSI objects.", !cqs1.equals(cqs2));
```
Comment (shahrs87) [inline phoenix-core/src/it/java/org/apache/phoenix/jdbc/PhoenixTestDriverIT.java:83]: You could have used assertNotEquals.

**Review by gjacoby126 — APPROVED** [review]
+1

**Pull request merged**